src/spi_drv.cpp

```diff
@@ -0,0 +1,341 @@
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL WI6LABS BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+-----------------------------------------------------------------------------*/
+
+
+#include <SPI.h>
+#include <stdbool.h>
+#include <string.h>
+#include "Arduino.h"
+#include "spi_drv.h"
+#include "pins_arduino.h"
+
+/**
+ * @brief  Constructor.
+ * @param  SPIx   : SPI interface
+ * @param  cs     : chip select pin
+ * @param  spiIRQ : SPI IRQ pin
+ * @param  reset  : reset pin
+ * @param  wakeup : wakeup pin
+ * @retval None
+ */
+SpiDrvClass::SpiDrvClass(SPIClass *SPIx, uint8_t cs, uint8_t spiIRQ,
+                         uint8_t reset, uint8_t wakeup) {
+  ISM43362 = SPIx;
+  csPin = cs;
+  spiIRQPin = spiIRQ;
+  wakeupPin = wakeup;
+  resetPin = reset;
+}
+
+/**
+ * @brief  Select the device.
+ * @param  None
+ * @retval None
+ */
+void SpiDrvClass::Spi_Slave_Select() {
+    digitalWrite(csPin,LOW);
+    delay(10);
+}
+
+/**
+ * @brief  Unselect the device.
+ * @param  None
+ * @retval None
+ */
+void SpiDrvClass::Spi_Slave_Deselect() {
+    digitalWrite(csPin,HIGH);
+    delay(10);
+}
+
+/**
+ * @brief  Get data ready state.
+ * @param  None
+ * @retval data_ready pin state. This is the spiIRQ pin.
+ */
+uint8_t SpiDrvClass::Spi_Get_Data_Ready_State()
+{
+  return digitalRead(spiIRQPin);
+}
+
+/**
+ * @brief  Reset the WiFi device.
+ * @param  None
+ * @retval None
+ */
+void SpiDrvClass::Spi_Wifi_Reset() {
+  digitalWrite(resetPin,LOW);
+  delay(10);
+  digitalWrite(resetPin, HIGH);
+  delay(500);
+}
+
+/**
+ * @brief  Initialize the SPI
+ * @param  None
+ * @retval : 0 if init success, -1 otherwise.
+ */
+int8_t SpiDrvClass::IO_Init(void)
+{
+  uint8_t Prompt[6];                     // data receive
+  uint8_t count = 0;                     // Number of bytes receive
+  uint32_t start;                        // start time for timeout
+  uint16_t dummy_send = 0x0A0A;          // Data to feed the clock
+  uint16_t read_value;                   // data read
+
+  /* pin configuration                                                       */
+  pinMode(wakeupPin, OUTPUT);
+  digitalWrite(wakeupPin, LOW);
+
+  pinMode(spiIRQPin, INPUT);
+
+  pinMode(resetPin, OUTPUT);
+
+  pinMode(csPin, OUTPUT);
+  digitalWrite(csPin, HIGH);
+
+  /* 80/8= 10MHz (Inventek WIFI module supportes up to 20MHz)                */
+  Settings_43362 = new SPISettings(SPI_CLOCK_DIV8, MSBFIRST, SPI_MODE0);
```

Comment:
Where the 80/8 comes from?
is it the SystemCoreClock  of the boards?
What happen if I change the clock config?

Reply:
Comes from the ST library. SPI_CLOCK_DIV8 will be replace by the frequency so nothing will happen if you change the clock config.